Build failure due to tag lock

Our build (`windows-2022` is terminating the other platforms) [is failing](https://github.com/Tribler/tribler/actions/runs/17543614482/job/49819914376) due to the following error:

```
error: cannot lock ref 'refs/tags/v7.1.5': Unable to create 'D:/a/tribler/tribler/.git/refs/tags/v7.1.5.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
Error: Process completed with exit code 255.
```

https://public-inbox.org/git/20250820213323.GA1667633@coredump.intra.peff.net/ and https://github.com/actions/runner-images/issues/12936 suggest this is due to case sensistivity and `git 2.51.0`. This seems to be correct.

The following output shows that we have, from a case-insensitive perspective, a duplicate tag:

```
 * [new tag]             V7.1.5                -> V7.1.5
 ...
 * [new tag]             v7.1.5                -> v7.1.5
```

Attempted fixes (note we use `fetch-depth: 1` already in the `checkout` Action):
 1. ❌ `git config core.ignorecase false` before `git fetch --tags`.
 2. ❌ `git fetch origin 'refs/tags/v*:refs/tags/v*'` instead of `git fetch --tags`: error changes to `error: cannot lock ref 'refs/tags/V7.3.0-beta2': Unable to create 'D:/a/tribler/tribler/.git/refs/tags/V7.3.0-beta2.lock': File exists.`.
 3. ❌ `git fetch --prune-tags --refetch --tags` instead of `git fetch --tags`.
 4. ✅ `git fetch --no-tags origin 'refs/tags/v*:refs/tags/v*'` instead of `git fetch --tags`. _Disclaimer_: this depends on our particular custom naming structure (i.e., all modern branches start with a lower-case 'v'), it is not a general solution for all repositories that have capitalization issues in their tags.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -17,6 +17,12 @@ jobs:
       ########################################
       # NSIS installer (Windows, always x86) #
       ########################################
+      - name: Install NSIS
+        if: matrix.os == 'windows-latest'
+        run: |
+          Invoke-WebRequest https://deac-riga.dl.sourceforge.net/project/nsis/NSIS%203/3.10/nsis-3.10.zip?viasf=1 -OutFile C:\WINDOWS\Temp\nsis-3.10.zip
+          Expand-Archive -LiteralPath "C:\WINDOWS\Temp\nsis-3.10.zip" -DestinationPath "C:\WINDOWS\Temp\unzip-nsis-3.10"
+          Move-Item -Path "C:\WINDOWS\Temp\unzip-nsis-3.10\nsis-3.10" -Destination "C:\Program Files (x86)\NSIS"
       - name: Check out SimpleFC repository (Windows)
         if: matrix.os == 'windows-latest'
         uses: actions/checkout@v4
@@ -42,6 +48,7 @@ jobs:
           msbuild nsProcess.vcxproj /p:PlatformToolset=v143 /p:Configuration="Release UNICODE"
           copy "Release UNICODE\nsProcess.dll" "C:\Program Files (x86)\NSIS\Plugins\x86-unicode\nsProcess.dll"
           cd ../Include
+          New-Item -ItemType Directory -Force -Path "C:\Program Files (x86)\NSIS\Include"
           copy nsProcess.nsh "C:\Program Files (x86)\NSIS\Include\nsProcess.nsh"
 
       #######################
@@ -103,7 +110,7 @@ jobs:
       - name: Determine release tag
         shell: bash
         run: |
-          git fetch --tags
+          git fetch --no-tags origin 'refs/tags/v*:refs/tags/v*'
           git for-each-ref --count=1 --sort=-creatordate --format '%(refname)' refs/tags > raw_tag.txt
           GITHUB_TAG=$(git name-rev --tags --name-only $(cat raw_tag.txt))
           echo "GITHUB_TAG=${GITHUB_TAG#v}" >> $GITHUB_ENV
@@ -173,6 +180,14 @@ jobs:
         run: |
           vcpkg install libsodium
           Copy C:/vcpkg/packages/libsodium_x64-windows/bin/libsodium.dll src/libsodium.dll
+      - name: Fetch old OpenSSL for libtorrent (Windows)
+        if: matrix.os == 'windows-latest'
+        run: |
+          Remove-Item -Path "C:\Program Files\OpenSSL" -Force -Recurse
+          Write-Output '{"$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json","dependencies": ["openssl"],"overrides": [{"name": "openssl","version-string": "1.1.1n"}]}' | Out-File -FilePath ".\vcpkg.json"
+          vcpkg x-update-baseline --add-initial-baseline
+          vcpkg install
+          New-Item -Path "C:\Program Files\OpenSSL" -ItemType SymbolicLink -Value "C:\vcpkg\packages\openssl_x64-windows\"
       - name: Build Executables (Windows)
         if: matrix.os == 'windows-latest'
         run: |

--- a/build/win/makedist_win.bat
+++ b/build/win/makedist_win.bat
@@ -48,7 +48,7 @@ REM --- Doing this in ugly way for now
 if not defined SKIP_SIGNING_TRIBLER_BINARIES (
     openssl req  -nodes -new -x509 -config build\win\keygen_config.txt -keyout key.pem -out pub_key.pem
     openssl pkcs12 -export -in pub_key.pem -inkey key.pem -out ot_cert.pfx -passout pass:
-    "C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64\signtool.exe" sign /f ot_cert.pfx /d "Tribler" /t "http://timestamp.digicert.com" dist\tribler\tribler.exe
+    "C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\x64\signtool.exe" sign /f ot_cert.pfx /fd SHA256 /d "Tribler" /t "http://timestamp.digicert.com" dist\tribler\tribler.exe
 )
 
 @echo Running NSIS
@@ -60,7 +60,7 @@ move Tribler_*.exe ..
 cd ..
 REM Arno: Sign installer
 if not defined SKIP_SIGNING_TRIBLER_BINARIES (
-    "C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64\signtool.exe" sign /f ..\ot_cert.pfx /d "Tribler" /t "http://timestamp.digicert.com" Tribler_*.exe
+    "C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\x64\signtool.exe" sign /f ..\ot_cert.pfx /fd SHA256 /d "Tribler" /t "http://timestamp.digicert.com" Tribler_*.exe
 )
 
 endlocal

--- a/doc/building/windows.rst
+++ b/doc/building/windows.rst
@@ -8,7 +8,7 @@ Additionally, you will need to install:
 - ``NSIS``, the ``SimpleFC`` plugin, and the ``nsProcess`` plugin.
 - The latest ``libsodium.dll`` release.
 - ``Microsoft Visual Studio 2022 Enterprise``. ``2022 Community`` will also work, but you need to edit `tribler.nsi` in the appropriate place.
-- ``Windows Kits 10.0.19041.0``.
+- ``Windows Kits 10.0.26100.0``.
 - ``OpenSSL``.
 
 .. note::

